src/batched/KokkosBatched_SolveUTV_TeamVector_Internal.hpp

```diff
@@ -142,39 +143,45 @@ namespace KokkosBatched {
 		   one,
 		   T, ts0, ts1,
 		   W, ws0, ws1);
+	member.team_barrier();
 	
 	/// X = V^T W
 	TeamVectorGemmInternal<Algo::Gemm::Unblocked>
 	  ::invoke(member,
-		   m, nrhs, matrix_rank, 
+		   n, nrhs, matrix_rank, 
 		   one,
 		   V, vs1, vs0,
 		   W, ws0, ws1,
 		   zero,
 		   X, xs0, xs1);
+	member.team_barrier();
       } else {
+	/// W = U^T B
 	TeamVectorGemmInternal<Algo::Gemm::Unblocked>
 	  ::invoke(member,
-		   m, nrhs, matrix_rank, 
+		   matrix_rank, nrhs, m, 
 		   one,
 		   U, us1, us0,
 		   B, bs0, bs1,
 		   zero,
 		   X, xs0, xs1);
+    member.team_barrier();
 
+	/// X = T^{-1} X
 	TeamVectorTrsmInternalLeftUpper<Algo::Trsm::Unblocked>
 	  ::invoke(member,
 		   false,
 		   matrix_rank, nrhs,
 		   one,
 		   T, ts0, ts1,
 		   X, xs0, xs1);		
+	member.team_barrier();
       }
       
       /// X = P^T X
       TeamVectorApplyPivotMatrixBackwardInternal
       	::invoke(member,
-      		 nrhs, m,
+      		 nrhs, n,
```

Comment:
Previously had matrix_rank here, which was incorrect and should have been n. However, this was never exposed in any of the problems that I tested on (since they tend to have the column that is rank deficient as the last one in most cases). Now passes the previously failing test, batched_scalar_teamvector_solve_utv2_double.

Reply:
Thanks @kuberry! I relaunched the spot-check testing